ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -233,6 +233,9 @@ func (r *RayClusterReconciler) rayClusterReconcile(request ctrl.Request, instanc
 			r.Log.Error(err, "Update status error", "cluster name", request.Name)
 		}
 	}
+	if instance.Status.State == rayiov1alpha1.Pending {
```

Comment:
In my understanding, the string member (i.e. `instance.Status.State`) in a struct (i.e. `instance`) will be deep copied by default [1]. Hence, `instance.Status.State = rayiov1alpha1.Pending` in the function `updateStatus` will not take effect here. That is, currently, this condition (`instance.Status.State == rayiov1alpha1.Pending`) will be fulfilled in the next reconciliation rather than this reconciliation if `r.Status().Update(...)` succeeds. On the other hand, if `r.Status().Update(...)` in the function `updateStatus` fails, it seems that the event will not be requeued again.

I did not run the code manually, so correct me if I miss any details. Thanks!

[1] https://echorand.me/posts/go-values-references-etc/#elements-in-maps-and-struct-types

Reply:
Good catch, I will double check this. It's possible my manual test passed because the reconciliation was called twice for unrelated reasons, which would have hidden this bug.

Reply:
Hi @kevin85421, in this case, `instance` is being passed as a pointer in `updateStatus`, so I think it's okay.  The blog post seems to agree with this.  What do you think?

Reply:
@kevin85421 Here `instance` is being passed to `updateStatus` as a pointer, so I think it should be okay. Is my understanding correct?

Reply:
I missed that it is passed as a pointer. So, this `if` condition will be fulfilled in the same reconciliation no matter `r.Status().Update(...)` succeeds or not. Thank you for pointing this out!